.flake8

```diff
@@ -0,0 +1,47 @@
+[flake8]
+################### FILE PATTERNS ##########################
+
+# Provide a comma-separated list of glob patterns to exclude from checks.
+exclude =
+    .git,
+    __pycache__,
+    .pytest_cache,
+    .mypy_cache,
+    .venv,
+    .vscode,
+    .idea,
+    test
+# Provide a comma-separate list of glob patterns to include for checks.
+filename =
+    *.py
+
+
+################### LINTING ################################
+# Report all errors, even if it is on the same line as a `# NOQA` comment.
+disable-noqa = False
+
+# Set the maximum length that any line (with some exceptions) may be.
+max-line-length = 120
+# Set the maximum allowed McCabe complexity value for a block of code.
+max-complexity = 10
+# Toggle whether pycodestyle should enforce matching the indentation of the opening bracket’s line.
+# incluences E131 and E133
+hang-closing = True
+
+ignore =
+    E128 # continuation line under-indented for visual indent
+    E133 # closing bracket is missing indentation
+    E201 # whitespace after '('
+    E225 # missing whitespace around operator
+    E231 # missing whitespace after ','
+    E241 #multiple spaces after ','
+    E252 #missing whitespace around parameter equals
+    E261 #at least two spaces before inline comment
+    E266 #too many leading '#' for block comment
+    E302 #expected 2 blank lines, found 1
+    E303 #too many blank line
+    E501 # line too long
+    F401 #'io' imported but unused
+    F841 #local variable is assigned to but never used
+    W293 #blank line contains whitespace
+    W391 #blank line at end of file
```

Comment:
Why we are ignoring these?
These seem exactly the sort of rules we should enforce.

Reply:
These rules have low value but high prevalence, so we would be swamped with many errors that when fixed do not contribute much to the cleanliness. Moreover, they could automatically be fixed using `autopep8` and `autoflake` libraries. I chose not to waste time now on them as they are mostly cosmetic. In addition, I did not want to mix the PR about adding the lining rules, with the PR of fixing these errors.